frontend/src/old-pages/Configure/Queues/__tests__/SubnetMultiSelect.test.tsx

```diff
@@ -0,0 +1,180 @@
+    a.every((val, index) => val === b[index])
+  )
+}
+
+describe('given a component to select multiple subnets in a VPC', () => {
+  describe('when a VPC has been specified', () => {
+    beforeEach(() => {
+      mockUseState.mockImplementation((path: string[]) => {
+        if (arrayEquals(path, ['app', 'wizard', 'vpc'])) {
+          return 'VPC-1'
+        } else {
+          return mockSubnets
+        }
+      })
+    })
+
+    it('should show as options only the subnets in that VPC', () => {
+      const mockedOnChange = jest.fn()
+      const {container} = render(
+        <SubnetMultiSelect
+          value={mockSelectedSubnets}
+          onChange={mockedOnChange}
+        />,
+      )
+
+      const multiSelect = wrapper(container).findMultiselect()
+      expect(multiSelect).toBeDefined()
+      multiSelect?.openDropdown()
+      expect(multiSelect?.findDropdown()?.findOptions().length).toBe(2)
+      expect(
+        multiSelect?.findDropdown()?.findOptionByValue('subnet-1'),
+      ).toBeTruthy()
+      expect(
+        multiSelect?.findDropdown()?.findOptionByValue('subnet-2'),
+      ).toBeTruthy()
+    })
+  })
+
+  describe('when only one subnet is selected', () => {
+    beforeEach(() => {
+      mockUseState.mockImplementation((path: string[]) => {
+        if (arrayEquals(path, ['app', 'wizard', 'vpc'])) {
+          return 'VPC-1'
+        } else {
+          return mockSubnets
+        }
+      })
+    })
+
+    it('should show the selected subnet only', () => {
+      const mockedOnChange = jest.fn()
+      const {container} = render(
+        <SubnetMultiSelect
+          value={mockSelectedSubnets}
+          onChange={mockedOnChange}
+        />,
+      )
+
+      const multiSelect = wrapper(container).findMultiselect()
+      expect(mockedOnChange).toHaveBeenCalledTimes(0)
+      expect(
+        multiSelect?.findToken(1)?.findLabel().getElement().textContent,
+      ).toBe('subnet-1')
+    })
+  })
+
+  describe('when no subnet is selected', () => {
+    beforeEach(() => {
+      mockUseState.mockImplementation((path: string[]) => {
+        if (arrayEquals(path, ['app', 'wizard', 'vpc'])) {
+          return 'VPC-1'
+        } else {
+          return mockSubnets
+        }
+      })
+    })
+
+    it('should not display any subnet', () => {
+      const mockedOnChange = jest.fn()
+      const {container} = render(
+        <SubnetMultiSelect value={[]} onChange={mockedOnChange} />,
+      )
+
+      const multiSelect = wrapper(container).findMultiselect()
+      expect(multiSelect?.findTokens().length).toBe(0)
+    })
+  })
+
+  describe('when the user selects multiple subnets', () => {
+    beforeEach(() => {
+      mockUseState.mockImplementation((path: string[]) => {
+        if (arrayEquals(path, ['app', 'wizard', 'vpc'])) {
+          return 'VPC-1'
+        } else {
+          return mockSubnets
+        }
+      })
+    })
+
+    it('should display the selected subnets', () => {
```

Comment:
question: what are you testing here? It seems to me you're just testing again a Cloudscape component behavior. What you can do instead is testing **your own** logic, for example the `onChange` passed to the `MultiSelect`

Reply:
You're right, in this way I am basically replicating CloudScape unit tests. Going to test that `setState` of my `onChange` is correctly called.